integers_and_arrays.rb

```diff
@@ -83,16 +111,51 @@ def intersection(array_1, array_2)
 # Assumption/ Given: All numbers in the matrix are 0s or 1s
 # If any number is found to be 0, the method updates all the numbers in the
 # corresponding row as well as the corresponding column to be 0.
-def matrix_convert_to_0(matrix)
-  puts "NOT IMPLEMENTED"
+def matrix_convert_to_0(matrix) # time complex O(n^2 + n^3) => O(n^3) (kind of, since that assumes n = m, which it doesnt), space complex O(2n) => O(n)
+  rows_with_0s = []
+  columns_with_0s = []
+  matrix.each_with_index do |row, x|
+    row.each_with_index do |item, y|
+      if item == 0
+        columns_with_0s << y
+        rows_with_0s << x
+      end
+    end
+  end
+  matrix.each_with_index do |row, x|
+    row.each_with_index do |item, y|
+      if rows_with_0s.include?(x) || columns_with_0s.include?(y)
+        matrix[x][y] = 0
+      end
+    end
+  end
+  return matrix
 end
```

Comment:
To further optimize space complexity, you could use matrix[i][0] where i ranges from 0 the (number of rows)-1 and matrix[0][j] where j ranges from 0 to (number of columns)-1 i.e. the first row and first column of the matrix to save information on which rows and which columns need to be made zero by simply making them 0 and later checking on it. This will make your algorithm have O(1) space complexity.